transactions/src/main/java/org/jboss/as/txn/subsystem/TransactionSubsystem14Parser.java

```diff
@@ -110,6 +116,11 @@ public void readElement(XMLExtendedStreamReader reader, List<ModelNode> list) th
             }
             readElement(reader, element, list, subsystem, logStoreOperation);
         }
+
+        if(!relativeToEncountered && relativeToHasDefaultValue) {
```

Comment:
So L68 becomes this:

protected boolean relativeToHasDefaultValue = true;

and then overridden in TransactionSubsystem31Parser? 

Yes, that's what we want.

If the parser generates object-store-relative-to=undefined and object-store-path=tx-object-store, then TransactionSubsystemAdd is going to substitute in jboss.server.data.dir anyway. So that suggests the parser doesn't need to do anything at all, since the add handler will do it.

But, I think TransactionSubsystemAdd, after L147, should log a WARN notifying the user what it did. But, if the op TransactionSubsystemAdd is processing came from this parser, and the document used an xsd version < 3.1, then we don't want that WARN, since the user just followed the xsd. So we want this code here to substitute in "jboss.server.data.dir" and thus prevent that WARN.

But we don't want it to do the substitution for 3.1 or later. We want TransactionSubsystemAdd to WARN in that case.